### PR TITLE
adding additional handles to sidebar

### PR DESCRIPTION
**Pull request by MuellerMatthew**
Hi Turbo,

I added several public and private functions to sidebar to give users more controls over the sidebar when they are using it.

New public functions:
 ``getActiveTab()`` Returns the currently shown sidebar tab.
``getDisabledTabs()`` Returns a list of the currently disabled tabs.
New private functions:
 ``_getActiveTab()`` Returns the currently shown sidebar tab(copied existing code which was in close function to create this function).
``_getDisabledTabs()`` Returns a list of the currently disabled tabs.

Modified the ``open`` function so it:
1. checks whether a tab is disabled before opening it, if the tab is disabled, it won't open.
2. checks if the id it is trying to open actually exists before opening the tab.  If the ID doesn't match it won't open anything. 

modified the ``close`` function so it uses the new ``_this.getActiveTab()`` code to simplify the function.

### CODE REVIEW

**Comment by MuellerMatthew** [comment]
FYI i will probably be sending another pull request soon with additional functions to:
1. disable/enable tabs
2. dynamically add/remove/copy tabs in the sidebar( this could be used for search results where there is a lot of information to generate a tab for each search result shown on the map, and you could use the ``getActiveTab`` function to determine which tab is show, so you can highlight the matching feature.).
3. add events such as `tabOpen` and `tabClose` so the map can listen to the visible tabs and change based on which one is visible.

**Comment by josephzhao** [comment]
if my page has top header with fixed top menubar, the sidebar will not display  correctly, is there any example for my situation ?

Thanks

**Comment by MuellerMatthew** [comment]
@josephzhao If you have questions you should open a new issue, instead of posting off topic. 

You need to modify your css for the sidebar.  All of my maps have a 60px banner on top for the title and additional features, so i have my sidebar vertically offset from the top of the screen 70px, so there is a 10px gap.  `.sidebar{ top: 70px}`. Depending on your css media queries, and whether your banner changes sizes, or scrolls, you will probably need to do some additional css changes to accommodate your situation, but you should be able to do everything with the css.

**Comment by MuellerMatthew** [comment]
Adding 3 additional functions:
`.getTab(<id>)` returns the tab or tabs with the matching id(s)
`.disable(<id>)` disables the tab with matching id.  if no id is passed, it disables all tabs.
`.enable(<id>)` enables the tab with matching id. if no id is passed, it enables all disabled tabs.
They can be passed:
```javascript
sidebar
     .enable("messages")
     .open("messages")
     .disable(["profile", "settings"]);
```

File: js/leaflet-sidebar.js [hunk 1]
```diff
@@ -136,6 +144,75 @@ L.Control.Sidebar = L.Control.extend({
...
+
+        for (i = id.length-1; i>=0; i--) {    
+            child = id[i].firstChild;
+            L.DomUtil.removeClass(child, 'disabled'); //if the tab is disabled, then enable it.                       
+       }
+        return this;
+    },
+    
+    getActiveTab: function() {
+        return this._getActiveTab();
```
Comment (Turbo87) [inline js/leaflet-sidebar.js]: whats the point of that private function if the public one does the same thing?

**Comment by Turbo87** [comment]
could you try and format the code like the already existing code please?

**Pull request closed without merging**